### ISSUE TITLE
Clickhouse preparestatement can't use insert into select 

### ISSUE THREAD

**Issue by wsg96321** [ClickHouse/clickhouse-jdbc#402]
when use command client,  
clickhouse-client --host test-machine, it is success when use 
`insert into test_part.l_test_log (uid,uuid) select 2,generateUUIDv4() `

But when use jdbc for this sql , it will get exception like 
`Query must be like 'INSERT INTO [db.]table [(c1, c2, c3)] VALUES (?, ?, ?)`

in java class ru.yandex.clickhouse.ClickHousePreparedStatementImpl
it it 

>    public int[] executeBatch(Map<ClickHouseQueryParam, String> additionalDBParams) throws SQLException {
        Matcher matcher = VALUES.matcher(sql);
        if (!matcher.find()) {
            throw new SQLSyntaxErrorException(
                    "Query must be like 'INSERT INTO [db.]table [(c1, c2, c3)] VALUES (?, ?, ?)'. " +
                            "Got: " + sql
            );
        }
        int valuePosition = matcher.start();
        String insertSql = sql.substring(0, valuePosition);
        BatchHttpEntity entity = new BatchHttpEntity(batchRows);
        sendStream(entity, insertSql, additionalDBParams);
        int[] result = new int[batchRows.size()];
        Arrays.fill(result, 1);
        batchRows = new ArrayList<byte[]>();
        return result;
    }
>

**Comment by wsg96321** [comment]
I have no idea to get method to insert a batch record by clickhouse auto generate uuid. I have try to use this jdbc and other two third party driver like ClickHouse-Native-JDBC and clickhouse4j, event http method is also can't do it . At last I use snowflake algorithm to generate uuid by myself before I generate the batch insert . It will increase about 13% time to insert and increase about 20% disk to use for my this produce data .

**Comment by aaime** [comment]
Been bitten by the same, but in my case, I need to generate new rows by averaging previous ones. It can be done by computing the averages in the client, but seems inefficient. 

The sample query looks as follow:

``sql
INSERT INTO act_rpix.S2_RPIX
SELECT ?, 
max(resolution) - 1, 
max(date), 
round(avg(B01)),
round(avg(B02)),
round(avg(B03)),
round(avg(B04)),
round(avg(B05)),
round(avg(B06)),
round(avg(B07)),
round(avg(B08)),
round(avg(B09)),
round(avg(B10)),
round(avg(B11)),
round(avg(B12)),
round(avg(B8A)),
avg(NDVI),
avg(NDBI),
avg(NDWI)
FROM S2_H3
WHERE zoneId in (?, ?, ?, ?, ?, ?)
GROUP BY resolution
HAVING count(*) = 6
``

In other words, it's averaging the values of 6 existing rows (representing rHealPix cells, a DGGS system like H3) to create a parent cell, but only if all 6 children are present.
Skipping prepared statements works, but it's sub-optimal. 

I'm guessing this is currently done because the number of updated rows by a "insert xxx select" cannot be computed? The code is just filling the result array with 1:
https://github.com/ClickHouse/clickhouse-jdbc/blob/master/src/main/java/ru/yandex/clickhouse/ClickHousePreparedStatementImpl.java#L325

**Comment by zhicwu** [comment]
Both cases work in new driver(v0.3.2+).
